shared/src/main/scala/org/adridadou/openlaw/oracles/StopContractOracle.scala

```diff
@@ -1,57 +1,33 @@
 package org.adridadou.openlaw.oracles
 
-import cats.implicits._
 import org.adridadou.openlaw.parser.template.variableTypes._
 import org.adridadou.openlaw.vm._
 import io.circe.{Decoder, Encoder}
-import org.adridadou.openlaw.parser.template.VariableName
 import io.circe.syntax._
 import io.circe.generic.semiauto._
-import org.adridadou.openlaw.result.{Failure, Result}
-
-case class StopContractOracle(crypto:CryptoService) extends OpenlawOracle[StopExecutionEvent] {
-
-  override def incoming(vm:OpenlawVm, event: StopExecutionEvent): Result[OpenlawVm] = {
-    val identityResult = vm
-      .getAllVariables(IdentityType)
-      .toList
-      .map { case (id,variable) => (variable.name, vm.evaluate[Identity](id, variable.name)) }
-      .collect {
-        // TODO: Figure out which errors are expected here and allow remaining failures to stop processing
-        case (id, Right(identity)) =>
-          vm.isSignatureValid(vm.contractDefinition.id(crypto).stopContract(crypto), OpenlawSignatureEvent(vm.contractId, identity.email, "", event.signature, EthereumHash.empty)).map { x =>
-            (id, identity) -> x
-          }
-      }
-      .sequence
-      .map { list =>
-        list.find {
-          case (_, boolean) => boolean
-        }
-      }
-
-    identityResult.flatMap {
-      case Some(((name:VariableName, identity:Identity), _)) => processEvent(vm, event, name.name, identity)
-      case _ => Failure("invalid event! no matching identity while stopping the contract")
-    }
-  }
+import org.adridadou.openlaw.result.Result
+
+case class StopContractOracle(crypto:CryptoService) extends SignedActionOracle[StopExecutionEvent] {
+
+  override def incoming(vm:OpenlawVm, event: StopExecutionEvent): Result[OpenlawVm] =
+    checkAction(vm, crypto, event, vm.contractDefinition.id(crypto).stopContract(crypto))
 
   override def shouldExecute(event: OpenlawVmEvent): Boolean = event match {
     case _:StopExecutionEvent => true
     case _ => false
   }
 
-  private def processEvent(vm:OpenlawVm, event:StopExecutionEvent, name: String, identity:Identity): Result[OpenlawVm] = {
+  def processEvent(vm:OpenlawVm, event:StopExecutionEvent, name: String, identity:Identity): Result[OpenlawVm] = {
     vm(UpdateExecutionStateCommand(ContractStopped, event))
   }
 }
 
 object StopExecutionEvent {
-  implicit val stopExecutionEventEnc:Encoder[StopExecutionEvent] = deriveEncoder[StopExecutionEvent]
-  implicit val stopExecutionEventDec:Decoder[StopExecutionEvent] = deriveDecoder[StopExecutionEvent]
+  implicit val stopExecutionEventEnc:Encoder[StopExecutionEvent] = deriveEncoder
+  implicit val stopExecutionEventDec:Decoder[StopExecutionEvent] = deriveDecoder
 }
 
-case class StopExecutionEvent(signature: EthereumSignature) extends OpenlawVmEvent {
+case class StopExecutionEvent(signature: EthereumSignature) extends SignedActionEvent {
```

Comment:
we should check on the backend to make sure stop / resume still work with these changes, I know there have been some bugs with those features in the past